kotlin/samples/hello-terminal/terminal-workflow/src/main/java/com/squareup/sample/helloterminal/terminalworkflow/TerminalWorkflowRunner.kt

```diff
@@ -92,41 +90,22 @@ private suspend fun runTerminalWorkflow(
   keyStrokes: Flow<KeyStroke>,
   resizes: Flow<TerminalSize>
 ): ExitCode = coroutineScope {
-  var input = TerminalInput(screen.terminalSize.toSize(), keyStrokes.asWorker())
-  val inputs = ConflatedBroadcastChannel(input)
+  val keyStrokesWorker = keyStrokes.asWorker()
+  val inputs = resizes
+      .map { size ->
+        screen.doResizeIfNecessary()
+            // If the terminal was resized since the last iteration, we need to notify the workflow.
+            ?.toSize()
+            ?: size
+      }
+      .distinctUntilChanged()
+      .map { TerminalInput(it, keyStrokesWorker) }
 
   // Use the result as the parent Job of the runtime coroutine so it gets cancelled automatically
   // if there's an error.
-  val result =
-    launchWorkflowIn(this, workflow, inputs.asFlow()) { renderingsAndSnapshots, outputs ->
-      val renderings = renderingsAndSnapshots.map { it.rendering }
-          .produceIn(this)
-
-      val resizesChannel = resizes.produceIn(this)
-
-      launch {
-        while (true) {
-          val rendering = selectUnbiased<TerminalRendering> {
-            resizesChannel.onReceive {
-              screen.doResizeIfNecessary()
-                  ?.let {
-                    // If the terminal was resized since the last iteration, we need to notify the
-                    // workflow.
-                    input = input.copy(size = it.toSize())
-                  }
-
-              // Publish config changes to the workflow.
-              inputs.send(input)
-
-              // Sending that new input invalidated the lastRendering, so we don't want to
-              // re-iterate until we have a new rendering with a fresh event handler. It also
-              // triggered a render pass, so we can just retrieve that immediately.
-              return@onReceive renderings.receive()
-            }
-
-            renderings.onReceive { it }
-          }
-
+  val result = launchWorkflowIn(this, workflow, inputs) { renderingsAndSnapshots, outputs ->
```

Comment:
def. think that calling these things `runner` rather than `result` is easier to understand.